enterprise/app/trends/drilldown_page.tsx

```diff
@@ -307,6 +321,45 @@ export default class DrilldownPageComponent extends React.Component<Props, State
     this.fetchInvocationList();
   }
 
+  handleHeatmapZoom(s?: HeatmapSelection) {
+    this.currentHeatmapSelection = undefined;
+    this.currentZoomFilters = s;
+    this.fetch();
+  }
+
+  handleClearZoom() {
+    this.currentHeatmapSelection = undefined;
+    this.currentZoomFilters = undefined;
+    this.fetch();
+  }
+
+  addZoomFiltersToQuery(
+    // A little bit of structural typing for TrendQuery and InvocationQuery
+    query: {
```

Comment:
nit: you could also extract an explicit type def to make the func param definition slightly less syntax-soupy